python/ambassador/envoy/v2/v2listener.py

```diff
@@ -249,6 +249,15 @@ def add_tcp_group(self, irgroup: IRTCPMappingGroup) -> None:
         else:
             # What matching Hosts do we have?
             for host in sorted(self.config.ir.get_hosts(), key=lambda h: h.hostname):
+                # They're asking for a hostname match here, which _cannot happen_ without
+                # SNI -- so don't take any hosts that don't have a TLSContext.
+
+                if not host.context:
+                    if self._log_debug:
+                        self.config.ir.logger.debug("V2Listener %s @ %s TCP %s: skip %s", 
+                                                    self.name, self.bind_to, group_host, host)
+                    continue
+                
```

Comment:
If you're not going to set up SNI, it's fundamentally nonsensical to try a host-based TCP match.